### PR TITLE
Extract V2G evidence from a variant distance to a gene TSS

### PR DESCRIPTION
**Pull request by ireneisdoomed**
**This PR builds up on #41, so I'd rather review it once the other one is merged**

This PR includes:
- Process to generate V2G from a variant distance to the canonical TSS of a gene
- Scoring of the dataset
- Tests and docs
- Changes to the target schema, as it uses one dataset where the target index has the canonical TSS calculated.

### CODE REVIEW

File: src/etl/v2g/distance/distance.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    Returns:
+        DataFrame: The V2G df with the normalised score column.
+    """
+    inverse_expr = (max_distance - f.col("distance")).alias("tmp_score")
+    return (
+        df.select("*", inverse_expr)
+        .transform(lambda df: normalise_column(df, "tmp_score", "new_score"))
+        .drop("tmp_score")
+    )
```
Comment (DSuveges) [inline src/etl/v2g/distance/distance.py:54]: Why not:
```python
    return (
        df.withColumn('tmp_score', max_distance - f.col("distance"))
        .transform(lambda df: normalise_column(df, "tmp_score", "new_score"))
        .drop("tmp_score")
    )
```
Also, I think the `normalise_column` might be written in a way that you could say:

```python
    return (
        df.withColumn('new_score', normalise_column(max_distance - f.col("distance")))
    )
```

File: src/etl/v2g/distance/distance.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+def get_variant_distance_to_gene(
+    gene_df: DataFrame,
+    variant_df: DataFrame,
+    distance_window: int,
+) -> DataFrame:
+    """Calculates the distance between a variant and the TSS of a gene."""
+    return (
+        variant_df.select("variantId", "chromosome", "position")
+        .join(f.broadcast(gene_df), on="chromosome", how="inner")
```
Comment (DSuveges) [inline src/etl/v2g/distance/distance.py:65]: I'm sure you have explored this, but I'm curious: if you join by the two dataframes by already specifying the condition (`f.abs(f.col("position") - f.col("tss")<= distance_window`) wouldn't be faster? The resulting dataframe would be way smaller and you would save on the filtering. But spark might optimize on the backend already.

**Review by DSuveges — CHANGES_REQUESTED** [review]
Mostly things makes sense, I have only a handful of comments. There are things to sort out in the run script that that part is under revision by David anyway.

File: src/run_v2g.py [hunk 3]
```diff
@@ -29,23 +30,34 @@ def main(cfg: DictConfig) -> None:
...
+        filter_types=list(cfg.etl.v2g.parameters.approved_biotypes),
+    ).persist()
     va = etl.read_parquet(
         cfg.etl.v2g.inputs.variant_annotation, "variant_annotation.json"
     ).selectExpr("id as variantId", "chromosome", "vep")
+
+    etl.logger.info("Generating V2G evidence from distance assesments...")
+    v2g_distance_df = v2g_distance(
+        etl,
+        vi,
```
[the remark this answers is not in the document]
Comment (d0choa) [reply src/run_v2g.py:51]: @ireneisdoomed might have picked this from me. I have done some of this in my branch when things like `vi` are objects of the class `VariantIndex` so it's really easy to know what's going on. I'm happy either way

[the remark this answers is not in the document]
Comment (DSuveges) [reply src/run_v2g.py:51]: Fine.

File: src/etl/v2g/distance/distance.py (hunk 2 above)
Comment (d0choa) [reply src/etl/v2g/distance/distance.py:65]: In main we have the same situation [here](https://github.com/opentargets/genetics_etl_python/blob/57b0adf9e68e10f4256826ea00f3b7c6de7dc915/src/etl/v2g/intervals/helpers.py#L42)

I don't know how smart Spark is with this things. But @DSuveges sounds like a better solution to me

File: src/etl/v2g/distance/distance.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame
+
+    from etl.common.ETLSession import ETLSession
+
+from etl.common.spark_helpers import normalise_column
+
+
+def main(
```
[the remark this answers is not in the document]
Comment (d0choa) [reply src/etl/v2g/distance/distance.py:17]: It's not worth spending time on this. I will get rid of it anyway the moment it goes to `do-hydra`. (I agree with @DSuveges BTW)

**Pull request merged**